Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/ReflectionAdapter.cpp

```diff
@@ -927,36 +1097,28 @@ namespace AZ::DocumentPropertyEditor
                 switch (action.value())
                 {
                 case ContainerAction::AddElement:
-                    containerEntry->OnAddElement(m_impl.get(), message.m_messageOrigin);
+                    containerEntry->container->OnAddElement(m_impl.get(), message.m_messageOrigin);
                     break;
                 case ContainerAction::RemoveElement:
-                    containerEntry->OnRemoveElement(m_impl.get(), message.m_messageOrigin);
+                    containerEntry->element->OnRemoveElement(m_impl.get(), message.m_messageOrigin);
                     break;
                 case ContainerAction::Clear:
-                    containerEntry->OnClear(m_impl.get(), message.m_messageOrigin);
+                    containerEntry->container->OnClear(m_impl.get(), message.m_messageOrigin);
```

Comment:
This should check that either the BoundContainer or ContainerElement pointers aren't nullptr.

Reply:
Done. Same for below in the associative container key handlers.